In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#--------------------------------------------------------

from sklearn.metrics import confusion_matrix
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score
from xgboost import XGBClassifier
from sklearn.svm import SVC


In [2]:
df = pd.read_csv("C://Users//admin//Desktop//Mobile_Classification//Data_Preprocessing//scaled_data.csv")
df.head()

ram  battery_power  int_memory  mob_weight  px_height  px_width  \
0  0.391703      -0.902597   -1.380644    1.349249  -1.408949 -1.146784   
1  0.467317      -0.495139    1.155024   -0.120059   0.585778  1.704465   
2  0.441498      -1.537686    0.493546    0.134244   1.392684  1.074968   
3  0.594569      -1.419319   -1.215274   -0.261339   1.286750  1.236971   
4 -0.657666       1.325906    0.658915    0.021220   1.268718 -0.091452   

   price_range  
0            1  
1            2  
2            2  
3            2  
4            1

In [3]:
x = df.drop(['price_range'] , axis = 1)
y = df.price_range

In [4]:
from sklearn.model_selection import train_test_split
x_train , x_test , y_train , y_test = train_test_split(x , y , test_size = 0.25 , random_state = 2)

#### --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### RandomForestClassifier

In [5]:
classifier = RandomForestClassifier()
classifier.fit(x_train , y_train)
print(classifier.score(x_test , y_test))

scores = cross_val_score(classifier , x , y , cv = 5)
print(scores.mean())

0.926
0.915


#### Confusion Matrix

In [6]:
y_pred = classifier.predict(x_test)
cm = confusion_matrix(y_test , y_pred)
cm

array([[118,   4,   0,   0],
       [ 13, 115,   5,   0],
       [  0,   5, 109,   5],
       [  0,   0,   5, 121]], dtype=int64)

#### --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### XGBoost

In [7]:
xgb = XGBClassifier()
xgb.fit(x_train , y_train)
print(xgb.score(x_test , y_test))
scores = cross_val_score(xgb , x , y , cv = 5)
print(scores.mean())

0.922
0.921


#### Confusion Matrix

In [8]:
y_pred = xgb.predict(x_test)
cm = confusion_matrix(y_test , y_pred)
cm

array([[120,   2,   0,   0],
       [ 13, 112,   8,   0],
       [  0,   5, 108,   6],
       [  0,   0,   5, 121]], dtype=int64)

#### --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### SVM

In [9]:
svm = SVC(kernel = 'linear')
svm.fit(x_train , y_train)
print(svm.score(x_test , y_test))
scores = cross_val_score(svm , x , y , cv = 5)
print(scores.mean())

0.974
0.9695


#### Confusion Matrix

In [10]:
y_pred = svm.predict(x_test)
cm = confusion_matrix(y_test , y_pred)
cm

array([[122,   0,   0,   0],
       [  8, 122,   3,   0],
       [  0,   0, 118,   1],
       [  0,   0,   1, 125]], dtype=int64)

#### --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### Overall SVM gives best accuracy with 96% - 97% accruracy

### test score :- 97

### cross validarion score :- 96

### Model does not need tuning because it is giving good accuracy

### with minimum type1 error and 1 type2 error.

### Hence SVM will be selected for prediction.